### PR TITLE
8253089: Windows (MSVC 2017) build fails after JDK-8243208

### PR DESCRIPTION
**Pull request by shipilev**
It seems that MSVC 2017 is getting confused about the differences in `unsigned int` and `u2`. After a few attempts at fixing this, I think we need to use `u2` consistently for hash code computations. `u2` is the final storage type for the hash code in `JVMFlagLookup::_hashes`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253089](https://bugs.openjdk.java.net/browse/JDK-8253089): Windows (MSVC 2017) build fails after JDK-8243208

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Goetz Lindenmaier](https://openjdk.java.net/census#goetz) (@GoeLin - **Reviewer**)
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/150/head:pull/150`
`$ git checkout pull/150`

### CODE REVIEW

**Comment by GoeLin** [comment]
Hi, 
this fix works in our CI. Looks good.
Best regards,
  Goetz.

**Review by TheRealMDoerr — APPROVED** [review]

**Review by GoeLin — APPROVED** [review]
LGTM, Goetz.

**Comment by iklam** [comment]
I agree with Kim that it's better to disable the warning.

**Comment by iklam** [comment]
TL/DR; I think the proposed change is acceptable as it doesn't significantly change the bucket size distribution.

====
I added a patch to show the bucket size distribution:
http://cr.openjdk.java.net/~iklam/jdk16/8253089-jvmFlagLookup-u2-msvc/buckets_size.patch.txt

I then apply the proposed change on top. The contents of the hashtable is changed (I don't know why Aleksey sees no changes; I used output from `gcc -save-temps`). Even the lower 16 bit of the hash may not change, this expression will give different results depending on the upper 16 bits:
```
int bucket_index = (int)(hash % NUM_BUCKETS);
```
Anyway, the bucket size distribution is about the same as before. The max bucket size is about 12.

http://cr.openjdk.java.net/~iklam/jdk16/8253089-jvmFlagLookup-u2-msvc/jvmFlagLookup.s.old.txt
http://cr.openjdk.java.net/~iklam/jdk16/8253089-jvmFlagLookup-u2-msvc/jvmFlagLookup.s.new.txt

Scroll down to here: the first 277 entries are the bucket size.

```
_ZL18_flag_lookup_table:
	.value	2
	.value	8
	.value	3
	.value	5
	.value	10
	.value	1
	.value	7
	.value	8
	.value	3
	.value	3
	.value	5
	.value	2
	.value	5
	.value	7
	.value	6
	.value	2
	.value	5
	.value	2
	.value	6
	.value	3
	.value	8
	.value	1
	.value	3
```

**Review by iklam — APPROVED** [review]

**Comment by shipilev** [comment]
I don't mind reverting this change, but it would break MSVC 2017 again, until we push the pragmas. Unfortunately, I lost the capability to build with MSVC 2017, so maybe we should ask SAP folks (who IIRC still have their pipelines with MSVC 2017) to verify this still works.
